Source/MLX/Transforms+Compile.swift

```diff
@@ -180,6 +180,27 @@ public func compile(
     }
 }
 
+/// Overload of ``compile(inputs:outputs:shapeless:_:)-7korq`` that takes three ``MLXArray`` and
+/// produces a single ``MLXArray``.
+///
+/// ### See Also
+/// - <doc:compilation>
+/// - ``compile(inputs:outputs:shapeless:_:)-7korq``
+public func compile(
+    inputs: [any Updatable] = [], outputs: [any Updatable] = [], shapeless: Bool = false,
+    _ f: @escaping (MLXArray, MLXArray, MLXArray) -> MLXArray
```

Comment:
A three argument form -- I thought I needed it but ended up not using it, but it may be useful.  Generally people can consume and produce any number of MLXArray.  This is just a convenient wrapper for marshaling and unmarshaling